src/core/api.proto

```diff
@@ -30,6 +30,47 @@ package nvidia.inferenceserver;
 
 //@@.. cpp:namespace:: nvidia::inferenceserver
 
+//@@      .. cpp:var:: message InferSharedMemoryElement
+//@@
+//@@         The meta-data for the shared memory from which to read the input
+//@@         data and/or write the output data.
+//@@
+message InferSharedMemoryElement
+{
+  //@@    .. cpp:var:: string shm_key
+  //@@
+  //@@    The name of the shared memory region that holds the input data
+  //@@    (or where the output data should be written).
+  //@@
+  string shm_key = 1;
+
+  //@@    .. cpp:var:: uint64 offset
+  //@@
+  //@@       The offset from the start of the shared memory region.
+  //@@       element0 start = offset0, end = offset0 + size;
+  //@@
+  uint64 offset = 2;
+
+  //@@    .. cpp:var:: uint64 byte_size
+  //@@
+  //@@       Size of the element in the batch of the input tensor, in bytes.
```

Comment:
"Size of the element, in bytes". Don't talk about batch or input since that is specific and this is a generic structure.